src/main/java/entity_layer/CommonTaskTree.java

```diff
@@ -42,7 +85,17 @@ public boolean removeSubTaskTree(TaskTree taskTree) {
      */
     @Override
     public boolean removeChildTaskTree(TaskTree taskTree) {
-        return false;
+        // Check if it's a direct subtree
+        if (this.removeSubTaskTree(taskTree))
```

Comment:
From this implementation, it's clear to me that the removeSubTaskTree method should be private because if you try this method, it will always call it; this functionally eliminates the need for someone outside of this class to ever use the removeSubTaskTree method because it is a less powerful helper method

Reply:
Also, there should probably be a way to search out and remove a TaskTree by id

Reply:
Very true, I hadn't caught that. I'll change it now.

Reply:
As for the ID bit, My thought is that the existing methods would allow an interactor to do that already. However, I may add this as an actual method later down the line and expand the entities capabilities.